mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -165,14 +223,20 @@ export const WayPage = (props: WayPageProps) => {
             isWayInFavorites && setUser &&
             <Button
               value={"Remove from favorite"}
-              onClick={() => deleteFavoriteWayUuid(user, way.uuid, setUser)}
+              onClick={() => {
+                deleteFavoriteWayUuid(user, way.uuid, setUser);
+                deleteFavoriteForUserUuids(way, user.uuid, setWay);
+              }}
             />
           }
           {
             !isWayInFavorites && user && setUser &&
             <Button
               value={"Add to favorite"}
-              onClick={() => addFavoriteWayUuid(user, way.uuid, setUser)}
+              onClick={() => {
+                addFavoriteWayUuid(user, way.uuid, setUser);
+                addFavoriteForUserUuids(way, user.uuid, setWay);
```

Comment:
Looks like it is two functions which trigger Dal layer and should be triggered in transaction or batching, otherwise we will get broken data sooner or later, what do you 💬 no about moving this logic to one specific function which will be responsible for adding uuid in all required models? Discuss it with Ekaterina . The same for delete

Reply:
Added batch.